tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -539,3 +542,314 @@ def estimate_affine_transform(
+                A binary tissue mask for the moving image.
+
+        Returns:
+            tuple:
+                - np.ndarray - A cropped image containing tissue region.
+                - np.ndarray - A cropped image containing tissue mask.
+                - np.ndarray - A cropped image containing tissue region.
+                - np.ndarray - A cropped image containing tissue mask.
+                - tuple - Bounding box (min_row, min_col, max_row, max_col).
+
+        """
+        if len(fixed_mask.shape) != 2:
+            fixed_mask = fixed_mask[:, :, 0]
+        if len(moving_mask.shape) != 2:
+            moving_mask = moving_mask[:, :, 0]
+
+        fixed_mask = np.uint8(fixed_mask > 0)
+        moving_mask = np.uint8(moving_mask > 0)
+
+        regions = regionprops(fixed_mask)
+        fixed_minr, fixed_minc, fixed_maxr, fixed_maxc = regions[0].bbox
+        regions = regionprops(moving_mask)
+        moving_minr, moving_minc, moving_maxr, moving_maxc = regions[0].bbox
+        minc, maxc, minr, maxr = (
+            np.min([fixed_minc, moving_minc]),
+            np.max([fixed_maxc, moving_maxc]),
+            np.min([fixed_minr, moving_minr]),
+            np.max([fixed_maxr, moving_maxr]),
+        )
+
+        fixed_tissue_image = fixed_image[minr:maxr, minc:maxc]
+        fixed_tissue_mask = fixed_mask[minr:maxr, minc:maxc]
+        moving_tissue_image = moving_image[minr:maxr, minc:maxc]
+        moving_tissue_mask = moving_mask[minr:maxr, minc:maxc]
+        moving_tissue_image[np.all(moving_tissue_image == (0, 0, 0), axis=-1)] = (
+            243,
+            243,
+            243,
+        )
+        return (
+            fixed_tissue_image,
+            fixed_tissue_mask,
+            moving_tissue_image,
+            moving_tissue_mask,
+            (minr, minc, maxr, maxc),
+        )
+
+    @staticmethod
+    def find_points_inside_boundary(mask: np.ndarray, points: np.ndarray):
+        """Find indices of points lying inside the boundary.
+
+        This function returns indices of points which are
+        enclosed by an area indicated by a binary mask.
+
+        Args:
+            mask (:class:`numpy.ndarray`):
+                A binary tissue mask
+            points (:class:`numpy.ndarray`):
+                 (N, 2) array of point coordinates.
+
+        Returns:
+            :class:`numpy.ndarray`:
+                Indices of points enclosed by a boundary.
+
+        """
+        kernel = np.ones((25, 25), np.uint8)
+        mask = cv2.dilate(mask, kernel, iterations=1)
+        bound_points, _ = cv2.findContours(mask, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+
+        included_points_ind = []
+        for bound_p in bound_points:
+            bound_p = path.Path(np.squeeze(bound_p))
+            ind = bound_p.contains_points(points).nonzero()
+            included_points_ind = np.hstack([included_points_ind, ind[0]])
+        return included_points_ind.astype(int)
+
+    def filtering_matching_points(
```

Comment:
Is there not something similar available in cv2 or sklearn? This should be available in existing libraries I guess.

Reply:
https://docs.opencv.org/4.x/dc/dc3/tutorial_py_matcher.html

Reply:
Explore this option in parallel and create a PR later.